community/kernel/src/main/java/org/neo4j/kernel/impl/store/kvstore/AbstractKeyValueStore.java

```diff
@@ -317,7 +322,13 @@ private long rotate( boolean force ) throws IOException
                     // won't close the state as it was before rotation began, which we're reverting to right here.
                     try ( LockWrapper ignored = writeLock( updateLock ) )
                     {
-                        state = rotation.markAsFailed();
+                        // Only mark as failed if we're still running, because if this store has been stopped
+                        // this rotation task will be the only option to close the preState and marking
+                        // this rotation as failed will prevent preState to be closed.
+                        if ( !stopped )
+                        {
+                            state = rotation.markAsFailed();
+                        }
```

Comment:
That won't happen since stopping while in Rotation state will throw exception and not reassign the state.